src/Core__Option.res

```diff
@@ -22,73 +22,83 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
 
+let flat = opt =>
+  switch opt {
+  | Some(v) => v
+  | None => None
+  }
+
 let filterU = (opt, p) =>
   switch opt {
-  | Some(x) as some if p(. x) => some
+  | Some(value) as some if p(. value) => some
   | _ => None
   }
 
-let filter = (opt, p) => filterU(opt, (. x) => p(x))
+let filter = (opt, p) => filterU(opt, (. value) => p(value))
 
 let forEachU = (opt, f) =>
   switch opt {
-  | Some(x) => f(. x)
+  | Some(value) => f(. value)
   | None => ()
   }
 
-let forEach = (opt, f) => forEachU(opt, (. x) => f(x))
+let forEach = (opt, f) => forEachU(opt, (. value) => f(value))
 
-let getExn = x =>
-  switch x {
-  | Some(x) => x
+let getExn = opt =>
+  switch opt {
+  | Some(value) => value
   | None => raise(Not_found)
   }
 
+let expect = (opt, message) =>
+  switch opt {
+  | Some(value) => value
+  | None => raise(Failure(message))
+  }
+
 external getUnsafe: option<'a> => 'a = "%identity"
 
 let mapWithDefaultU = (opt, default, f) =>
   switch opt {
-  | Some(x) => f(. x)
+  | Some(value) => f(. value)
   | None => default
   }
 
-let mapWithDefault = (opt, default, f) => mapWithDefaultU(opt, default, (. x) => f(x))
+let mapWithDefault = (opt, default, f) => mapWithDefaultU(opt, default, (. value) => f(value))
 
 let mapU = (opt, f) =>
   switch opt {
-  | Some(x) => Some(f(. x))
+  | Some(value) => Some(f(. value))
   | None => None
   }
 
-let map = (opt, f) => mapU(opt, (. x) => f(x))
+let map = (opt, f) => mapU(opt, (. value) => f(value))
 
 let flatMapU = (opt, f) =>
   switch opt {
-  | Some(x) => f(. x)
+  | Some(value) => f(. value)
   | None => None
   }
 
-let flatMap = (opt, f) => flatMapU(opt, (. x) => f(x))
+let flatMap = (opt, f) => flatMapU(opt, (. value) => f(value))
 
 let getWithDefault = (opt, default) =>
   switch opt {
-  | Some(x) => x
+  | Some(value) => value
   | None => default
   }
 
-let orElse = (opt, other) =>
+let or = (opt, other) =>
   switch opt {
-  | Some(_) as some => some
+  | Some(_) => opt
   | None => other
   }
 
-let isSome = x =>
-  switch x {
-  | Some(_) => true
-  | None => false
-  }
+let orElse = or
+
+let isSome = opt => opt !== None
```

Comment:
Why does this use `!==`?

Reply:
The one case where this makes a difference is `None` vs `Some(null)`

Reply:
> Why does this use `!==`?

It just produces simpler code. `!==` compiles to that in JavaScript, while `!=` becomes a call to `Caml_obj,notequal`.

> The one case where this makes a difference is `None` vs `Some(null)`

Can't see that it does Or with `Some(undefined)`, which seems to me more likely to be problematic since `None` compiles to `undefined`.

Should probably add some tests for this, but are there cases that simple unit tests like `Option.isSome(Some(null))` wouldn't cover?

Reply:
Ignore my comment. Indeed the builtin `!=` distinguishes `null` and `undefined`. But here we want to avoid it as it is slower.